README
This script check if a host or IP is responding depending on the users inputs


CHANGELOG.txt
~~~~~~~~~~~~~~~~
version 0.1.5
~~~~~~~~~~~~~~~~

* Add https to github repository pulling
* Add the --update option to update the script when new feature available or any update

~~~~~~~~~~~~~~~~
version 0.1.4
~~~~~~~~~~~~~~~~

* Added the capability of running with python2 and calling directly using ./hping.py
* Fixed the problem when you don't enter a number for sleep, repeats, etc.

~~~~~~~~~~~~~~~~
version 0.1.3
~~~~~~~~~~~~~~~~

* Fix the error when no input is detected to use defaults
* Remove useless code or code that does not print out

~~~~~~~~~~~~~~~~
version 0.1.2
~~~~~~~~~~~~~~~~

* Fix some errors in the code

~~~~~~~~~~~~~~~~
version 0.1.1
~~~~~~~~~~~~~~~~

* initial release of HPing


check-ip.py
import os, sys
from time import sleep
import subprocess

# Give some color to the texts
class bcolors:
	GREEN = '\033[92m'
	YELLOW = '\033[93m'
	RED = '\033[91m'
	BOLD = '\033[1m'
	ENDC = '\033[0m'



def banner():
	pipinger = '''
 888888ba  oo  888888ba  oo                                     
 88    `8b     88    `8b                                        
a88aaaa8P' dP a88aaaa8P' dP 88d888b. .d8888b. .d8888b. 88d888b. 
 88        88  88        88 88'  `88 88'  `88 88ooood8 88'  `88 
 88        88  88        88 88    88 88.  .88 88.  ... 88       
 dP        dP  dP        dP dP    dP `8888P88 `88888P' dP       
oooooooooooooooooooooooooooooooooooooo~~~~.88~oooooooooooooooooo
                                      d8888P                    
	'''
	return pipinger

def check(address, get_alerts):
	try:
		with open(os.devnull, "wb") as limbo:
			ip=address
			result=subprocess.Popen(["ping", "-c", "1", "-n", "-W", "2", ip],
			stdout=limbo, stderr=limbo).wait()
			if result != 0:
				print(ip + bcolors.RED + bcolors.BOLD + " is not active"+ bcolors.ENDC)
			if result == 0 and get_alerts == True:
				print(ip + bcolors.GREEN + bcolors.BOLD + " is active"+ bcolors.ENDC)
	except Exception as error:
		print("An error ocurred: ")
		print(error)
		input("Press enter to continue...")
		main()

def main():
	try:
		os.system("clear")
		print(bcolors.YELLOW + bcolors.BOLD + banner() + bcolors.ENDC)
		address = input("Enter the IP/host you want to check: ")
		rest = int(input("How long do you want to wait until the next check? (between 0 and 60) "))
		if rest < 0 or rest > 60:
			print("Enter a number between 0 and 60")
			sys.exit()

		repeat = int(input("How many times would you like to check? (greater than 4, default is 200) "))
		x = 0
		
		alerts = input("Would you like to get alert if the host/IP is responding to pings? (Y/n) ").lower()
		if alerts == 'y':
			get_alerts = True
		else:
			get_alerts = False
		
		if repeat < 4:
			print("You entered %s which is not a valid input" % str(repeat))
			print("Try again and use a number between 4 and 1000")
			sys.exit()
		elif repeat >= 4 and repeat <= 1000:
			if get_alerts == False:
				print("\nYou will not see any results until the host or IP entered stops responding.")

			for i in range(0,repeat+1):
				x += 1
				if x == repeat+1:
					print("Done checking.\nRun the script again to check another host/ip")
				else:
					check(address, get_alerts)
					sleep(rest)
		else:
			repeat = 200
			x = 0
			if get_alerts == False:
				print("\nYou will not see any results until the host or IP entered stops responding.")

			for i in range(0,repeat+1):
				x += 1
				if x == repeat:
					print("Done checking.\nRun the script again to check another host/ip")
				else:
					check(address, get_alerts)
					sleep(rest)

	except KeyboardInterrupt:
		print("Caught ctrl+c\nExiting now...")
		sys.exit()
	except Exception as error:
		print("*"*20)
		print(error)
		print("*"*20)

main()


hping.py
#!/usr/bin/env python3

"""
HPing
Written by: Andrew (aka AgentWhite)
Twitter: @_agentwhite_
Website: https://thegibson.xyz
"""

version = "0.1.6"

import os, sys
from time import sleep
import subprocess

# python 3 compat
try: 
	input = raw_input
except NameError: 
	pass

# Give some color to the texts
class bcolors:
	GREEN = '\033[92m'
	YELLOW = '\033[93m'
	RED = '\033[91m'
	BOLD = '\033[1m'
	ENDC = '\033[0m'

def self_update():
	# force https for git
	def git_https_force():
		subprocess.Popen('git config --global url."https://github.com/".insteadOf git@github.com:;git config --global url."https://".insteadOf git://', shell=True).wait()

	# force https
	git_https_force()

	# try to update ourself first
	print("Trying to update myself first.. Then starting...")
	sleep(1)
	subprocess.Popen("git pull", shell=True).wait()

def banner():
	banner = bcolors.YELLOW + '''
ooooo ooooo oooooooooo  o88                           
 888   888   888    888 oooo  oo oooooo     oooooooo8 
 888ooo888   888oooo88   888   888   888  888    88o  
 888   888   888         888   888   888   888oo888o  
o888o o888o o888o       o888o o888o o888o 888     888 
                                           888ooo888                 
	''' + bcolors.ENDC
	banner += bcolors.BOLD + bcolors.RED + 'Version %s' % str(version)
	banner += '''
	HPing
	Written by: Andrew (aka AgentWhite)
	Twitter: @_agentwhite_
	Website: https://thegibson.xyz\n''' + bcolors.ENDC
	return banner

def check(address, get_alerts, count, repeat):
	try:
		with open(os.devnull, "wb") as limbo:
			ip=address
			result=subprocess.Popen(["ping", "-c", "1", "-n", "-W", "2", ip],
			stdout=limbo, stderr=limbo).wait()
			if result != 0:
				print("[" + bcolors.BOLD + bcolors.GREEN + str(count) + bcolors.ENDC + "/" + \
					bcolors.BOLD + bcolors.GREEN + str(repeat) + bcolors.ENDC + "] " + \
					bcolors.YELLOW + ip + bcolors.ENDC +bcolors.RED + bcolors.BOLD + " is not active"+ bcolors.ENDC)
			if result == 0 and get_alerts == True:
				print("[" + bcolors.BOLD + bcolors.GREEN + str(count) + bcolors.ENDC + "/" + \
				bcolors.BOLD + bcolors.GREEN + str(repeat) + bcolors.ENDC + "] " + bcolors.YELLOW \
				+ ip + bcolors.ENDC +bcolors.GREEN + bcolors.BOLD + " is active"+ bcolors.ENDC)
	except Exception as error:
		print("An error ocurred: ")
		print(error)
		input("Press enter to continue...")
		main()

def run_hping(address, get_alerts, repeat, rest):
	try:
		# Declare variables
		count = 0
		x = 0
		if get_alerts == False:
			print("\nYou will not see any results until the host or IP entered stops responding.")

		for i in range(0,repeat+1):
			x += 1
			count = x
			if count == repeat+1:
				print("Done checking.\nRun the script again to check another host/ip")
				sys.exit()
			else:
				check(address, get_alerts, count, repeat)
				sleep(rest)
	except KeyboardInterrupt:
		print("Caught ctrl+c\nExiting now...")
		sys.exit()
	except Exception as error:
		print("*"*20)
		print(error)
		print("*"*20)
		input("Press enter to continue...")
		main()

def main():
	try:
		os.system("clear")
		# Clear variables for when executing script again after exceptions
		address = ""
		rest = 0
		repeat = 0
		alerts = False
		
		print(banner())
		
		if '-h' in sys.argv or '--help' in sys.argv:
			print("\nJust run the program either python2 or python3\nYou can also run it using ./hping.py\n")
			sys.exit()
		
		address = input("Enter the IP/host you want to check: ")
		rest = input("How long do you want to wait until the next check? (default 4) ")
		if rest < '0' or rest > '60':
			print("Using default timeout as 4")
			rest = 4
		elif len(rest) <= 0:
			print("Using default timeout as 4")
			rest = 4
		
		repeat = input("How many times would you like to check? (greater than 4, default is 10) ")
		if repeat < '0':
			print("Using default to check every 60 seconds")
			repeat = 10
		elif len(repeat) == 0:
			print("Using default to check every 60 seconds")
			repeat = 10
			
		alerts = input("Would you like to get alert if the host/IP is responding to pings? (Y/n) ").lower()

		if alerts == 'y':
			get_alerts = True
		elif alerts == 'n':
			get_alerts = False
		else:
			get_alerts = True
		
		os.system("cls || clear")
		print(banner())
		repeat = int(repeat)
		rest = int(rest)
		
		run_hping(address, get_alerts, repeat, rest)

	except KeyboardInterrupt:
		print("Caught ctrl+c\nExiting now...")
		sys.exit()
	except Exception as error:
		print("*"*20)
		print(error)
		print("*"*20)

if __name__ == "__main__":
	if os.geteuid() != 0:
		print("\nWe autocheck for updates and need sudo permission since you most likely\ndownloaded me using PTF? IDK")
		print("This needs to be run as root. Please sudo it up! Exiting...")
		sys.exit()

	if '--update' in sys.argv:
		self_update()
		sleep(1)
		main()
	else:
		main()
